packages/edit-site/src/components/post-list/index.js

```diff
@@ -59,10 +63,28 @@ function useView( postType ) {
 	const [ view, setView ] = useState( selectedDefaultView );
 
 	useEffect( () => {
```

Comment:
What's the purpose of this effect?

Reply:
What happens if we just remove the effect?

Reply:
Ok I'm guessing this hook is about updating the "view" object when the url changes (probably for default views only). Then:

I'd suggest just using `setView( previous => { some merge here } )` and only update the stuff that come from the url rather than setting the whole object. It also avoids all these usePrevious...

Reply:
I think we probably don't need to compute the `selectedDefaultView` variable at all. We could just initialize the `useState` directly and update this hook to rely on the location params directly as well.

Reply:
Hi @youknowriad, this effect is what changes the view when the URL goes from `?postType=page&layout=grid&activeView=published` to `?postType=page&layout=grid&activeView=future`. The sidebar does not changes the view directly. The sidebar, just updates URL's. If we remove this effect the sidebar does not works at all.
If we agree on this solution or something similar I will add a comment with that on the effect to make it clear for the future.

Reply:
I've prepared an alternative PR at https://github.com/WordPress/gutenberg/pull/63889